pkg/cloudprovider/routecontroller/routecontroller.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package routecontroller
+
+import (
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/cloudprovider"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/golang/glog"
+)
+
+type RouteController struct {
+	routes      cloudprovider.Routes
+	kubeClient  client.Interface
+	clusterName string
+	clusterCIDR *net.IPNet
+}
+
+const k8sNodeRouteTag = "k8s-node-route"
+
+func New(routes cloudprovider.Routes, kubeClient client.Interface, clusterName string, clusterCIDR *net.IPNet) *RouteController {
+	return &RouteController{
+		routes:      routes,
+		kubeClient:  kubeClient,
+		clusterName: clusterName,
+		clusterCIDR: clusterCIDR,
+	}
+}
+
+func (rc *RouteController) Run(syncPeriod time.Duration) {
+	go util.Forever(func() {
+		if err := rc.reconcileNodeRoutes(); err != nil {
+			glog.Errorf("Couldn't reconcile node routes: %v", err)
+		}
+	}, syncPeriod)
+}
+
+func (rc *RouteController) reconcileNodeRoutes() error {
+	routeList, err := rc.routes.ListRoutes(rc.truncatedClusterName() + "-.*")
+	if err != nil {
+		return fmt.Errorf("error listing routes: %v", err)
+	}
+	nodeList, err := rc.kubeClient.Nodes().List(labels.Everything(), fields.Everything())
```

Comment:
// TODO: use pkg/controller/framework.NewInformer to watch this and reduce the number of lists needed.